src/ts/application.ts

```diff
@@ -0,0 +1,416 @@
+import { createLogger, logSection } from "./core/logging";
+import { StateManager } from "./core/state_manager";
+import { TrackedState } from "./core/tracked_state";
+import { getPlatformName, waitNextFrame } from "./core/utils";
+import { Vector } from "./core/vector";
+import { AdProviderInterface } from "./platform/ad_provider";
+import { NoAdProvider } from "./platform/ad_providers/no_ad_provider";
+import { NoAchievementProvider } from "./platform/browser/no_achievement_provider";
+import { AnalyticsInterface } from "./platform/analytics";
+import { GoogleAnalyticsImpl } from "./platform/browser/google_analytics";
+import { SoundImplBrowser } from "./platform/browser/sound";
+import { PlatformWrapperImplBrowser } from "./platform/browser/wrapper";
+import { PlatformWrapperImplElectron } from "./platform/electron/wrapper";
+import { PlatformWrapperInterface } from "./platform/wrapper";
+import { ApplicationSettings } from "./profile/application_settings";
+import { SavegameManager } from "./savegame/savegame_manager";
+import { AboutState } from "./states/about";
+import { ChangelogState } from "./states/changelog";
+import { InGameState } from "./states/ingame";
+import { KeybindingsState } from "./states/keybindings";
+import { MainMenuState } from "./states/main_menu";
+import { MobileWarningState } from "./states/mobile_warning";
+import { PreloadState } from "./states/preload";
+import { SettingsState } from "./states/settings";
+import { ShapezGameAnalytics } from "./platform/browser/game_analytics";
+import { RestrictionManager } from "./core/restriction_manager";
+import { PuzzleMenuState } from "./states/puzzle_menu";
+import { ClientAPI } from "./platform/api";
+import { LoginState } from "./states/login";
+import { WegameSplashState } from "./states/wegame_splash";
+import { MODS } from "./mods/modloader";
+import { MOD_SIGNALS } from "./mods/mod_signals";
+import { ModsState } from "./states/mods";
+
+export type AchievementProviderInterface =
+    import("./platform/achievement_provider").AchievementProviderInterface;
+
+export type SoundInterface = import("./platform/sound").SoundInterface;
+
+export type StorageInterface = import("./platform/storage").StorageInterface;
+
+const logger = createLogger("application");
+
+// Set the name of the hidden property and the change event for visibility
+let pageHiddenPropName, pageVisibilityEventName;
+if (typeof document.hidden !== "undefined") {
+    // Opera 12.10 and Firefox 18 and later support
+    pageHiddenPropName = "hidden";
+    pageVisibilityEventName = "visibilitychange";
+    // @ts-ignore
+} else if (typeof document.msHidden !== "undefined") {
+    pageHiddenPropName = "msHidden";
+    pageVisibilityEventName = "msvisibilitychange";
+    // @ts-ignore
+} else if (typeof document.webkitHidden !== "undefined") {
+    pageHiddenPropName = "webkitHidden";
+    pageVisibilityEventName = "webkitvisibilitychange";
+}
+
+export class Application {
+    public unloaded: boolean;
```

Comment:
I could not find where this is used, apart from setting it to false. If it is not used it can be removed